sdjournal/read.go

```diff
@@ -245,6 +253,13 @@ func (r *JournalReader) buildMessage() (string, error) {
 	var usec uint64
 	var err error
 
+	if r.formatter != nil {
+		entry, err := r.journal.GetEntry()
```

Comment:
Given that we are introducing a formatting function, I would suggest to lift the entry fetching logic into the common path at `Read`, then change the `buildMessage` signature to match the formatter one, and just call `r.formatter(entry)` there.